### PR TITLE
Short syntax for verb in GroupUrlRule

### PR DESCRIPTION
**Pull request by developeruz**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #7996

Duplicate of #9473 because the author is inactive.

### CODE REVIEW

File: framework/web/GroupUrlRule.php [hunk 1]
```diff
@@ -90,9 +90,16 @@ protected function createRules()
         $rules = [];
         foreach ($this->rules as $key => $rule) {
             if (!is_array($rule)) {
+                $verbs = 'GET|HEAD|POST|PUT|PATCH|DELETE|OPTIONS';
+                $verb = null;
+                if (preg_match("/^((?:($verbs),)*($verbs))\\s+(.*)$/", $key, $matches)) {
```
Comment (samdark) [inline framework/web/GroupUrlRule.php]: There's no need to capture what's not used:

```
^((?:(?:$verbs),)*(?:$verbs))\s+(.*)$
```

Then verb will be in $matches at index 1 and the rest at 2.

**Review by samdark — APPROVED** [review]
Overall looks good. Regex could be tweaked a bit.

File: framework/web/GroupUrlRule.php (hunk 1 above)
Comment (developeruz) [reply framework/web/GroupUrlRule.php]: Changed, thank you

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**